vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -0,0 +1,785 @@
+#include "conditional_contextual_bandit.h"
+#include "reductions.h"
+#include "example.h"
+#include "global_data.h"
+#include "cache.h"
+#include "vw.h"
+#include "interactions.h"
+#include "label_dictionary.h"
+#include "cb_adf.h"
+#include "cb_algs.h"
+
+#include <numeric>
+#include <algorithm>
+#include <unordered_set>
+#include <unordered_map>
+
+using namespace LEARNER;
+using namespace VW;
+using namespace VW::config;
+
+
+struct ccb
+{
+  example* shared;
+  std::vector<example*> actions, decisions;
+  uint32_t chosen_action_index;
+  std::unordered_map<uint32_t, uint32_t> origin_index;
+  CB::cb_class cb_label, default_cb_label;
+  std::unordered_set<uint32_t> exclude_list, include_list;
```

Comment:
Given the values in this set are known to be in the `[0 - data.actions.size() [`  range, this should be a bit-field such as ` std::vector<bool>`.